### ISSUE TITLE
Signature/Doc parser assumes <br> is present

### ISSUE THREAD

**Issue by grking** [DamnWidget/anaconda#427]
I'm using ST3, PyQt5, Python 3.  When typing a PyQT5 method name the signature popup doesn't appear and the following error appears in the ST3 console:

````python
ERROR:root:'' is not in list
ERROR:root:Traceback (most recent call last):
ERROR:root:  File "/home/gray/.config/sublime-text-3/Packages/Anaconda/anaconda_lib/jsonclient.py", line 94, in process_message
ERROR:root:    callback(data)
ERROR:root:  File "/home/gray/.config/sublime-text-3/Packages/Anaconda/listeners/signatures.py", line 58, in prepare_data
ERROR:root:    i = data['doc'].split('<br>').index("")
ERROR:root:ValueError: '' is not in list
````

This seems to be because PyQT5 doesn't return ````<br>```` in it's output, and Anaconda assumes it will.

Output from debug json server:
````python
About push back to ST3: b'{"success": true, "doc": "PyQt5.QtWidgets.QGraphicsScene.addItem\\nQGraphicsScene.addItem(QGraphicsItem)", "uid": "4f2e1776f47c4cd6a58dbc57b0487116"}\r\n'
````

**Comment by DamnWidget** [comment]
Oh but it has to be present https://github.com/DamnWidget/anaconda/blob/42cee33b1603a21855e5776f63a64021d8c8e4e7/anaconda_server/commands/doc.py#L46

Can you tell me about what configuration do you have for your `enable_signatures_tooltip`?

**Comment by grking** [comment]
Attached

[Anaconda.sublime-settings.txt](https://github.com/DamnWidget/anaconda/files/121451/Anaconda.sublime-settings.txt)

**Comment by DamnWidget** [comment]
That's so weird...

**Comment by DamnWidget** [comment]
I am not able to replicate myself
![screenshot_20160208_120011](https://cloud.githubusercontent.com/assets/329766/12885193/d63fec3c-ce5b-11e5-974a-4a1cd30e5110.png)

**Comment by grking** [comment]
Hmm.  Perhaps it's the PyQt5 install on Arch Linux.  I don't even get that method description if I do:

````python
Python 3.5.1 (default, Dec  7 2015, 12:58:09) 
[GCC 5.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from PyQt5.QtWidgets import QGraphicsScene
>>> help(QGraphicsScene.addItem)

Help on built-in function addItem:

addItem(...)
    QGraphicsScene.addItem(QGraphicsItem)
````

**Comment by DamnWidget** [comment]
I don't think so as I am using Manjaro (Arch based) and installed it just with pacman and everything looks fine, try to reinstall the package.

**Comment by grking** [comment]
I downloaded and installed PyQt5 from source.  Still the same.

I did just notice that the popup docstring you have does not relate to the addItem() method.  "Implement the central Checker class".  That is nothing to do with addItem().  In fact, that string does not appear anywhere in the PyQt5 download (after a configure and make):

````bash
[gray@box PyQt-gpl-5.5.1]$ grep -ri "Implement the central checker class" *
[gray@box PyQt-gpl-5.5.1]$ 
````

No matches.

So perhaps it's not my install that is odd for not having the doc string for that method, but your install is odd for having spurious doc strings? :)

In fact, let's test...

````bash
$ cd /usr
$ grep -ri "Implement the central checker class" *

lib/python3.5/site-packages/pyflakes/checker.py:Implement the central Checker class.
````

Interesting...

**Comment by DamnWidget** [comment]
I did check it again and, in fact it doesn't have any documentation at all (but still don't get the exception in your log).

![pyqt2](https://cloud.githubusercontent.com/assets/329766/12944066/e0c6e3ea-cfdc-11e5-9493-cf6c62899f04.png)

The docstring in the first screenshot has to be some residual string that was cached

**Comment by GriMel** [comment]
I've experienced the same bug.
ST3, 3083 and using PyQt4

**Comment by grking** [comment]
Appears fixed for me after upgrade to Anaconda 1.4.18 today.  Fixed by 387e31a2e8db91

**Comment by DamnWidget** [comment]
Closing it
